src/main/java/seedu/address/logic/parser/task/MarkTaskCommandParser.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.logic.parser.task;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.task.MarkTaskCommand;
+import seedu.address.logic.parser.ArgumentMultimap;
+import seedu.address.logic.parser.ArgumentTokenizer;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.TaskParserUtil;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new MarkTaskCommand object
+ */
+public class MarkTaskCommandParser implements Parser<MarkTaskCommand> {
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddCommand
+     * and returns an AddCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public MarkTaskCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args);
+
+        Index task_index;
+        Index person_index;
+
+        try {
+            task_index = TaskParserUtil.parseIndex(argMultimap.getPreamble());
+            person_index = TaskParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException ive) {
```

Comment:
Maybe catching ParseException will be more precise here since parseIndex throws ParseException.

Reply:
Noted.

Reply:
Finished.